java/ql/src/semmle/code/java/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1548,42 +1560,32 @@ private predicate flowIntoCallable(
   )
 }
 
-/** Holds if data may flow from `p` to a return statement in the callable. */
+/** Holds if data may flow from `p` to a return at position `pos`. */
 pragma[nomagic]
-private predicate paramFlowsThrough(ParameterNode p, CallContextCall cc, Configuration config) {
-  exists(PathNodeMid mid, ReturnNode ret |
-    mid.getNode() = ret and
+private predicate paramFlowsThrough(
+  ParameterNode p, ReturnPosition pos, CallContextCall cc, Configuration config
+) {
+  exists(PathNodeMid mid |
+    mid.getNode() = pos.getAReturnNode() and
     cc = mid.getCallContext() and
-    config = mid.getConfiguration() and
-    mid.getAp() instanceof AccessPathNil
```

Comment:
Hmm... something is fishy here. If we're tracking a field, we'll need to know that the access path was unchanged in order to summarise the flow, i.e. we cannot conflate taint-flows-through and value-flows-through.

Reply:
Good catch!